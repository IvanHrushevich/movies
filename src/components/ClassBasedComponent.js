import React from "react";

export class ClassBasedComponent extends React.Component {
  render() {
    return <div>Hello from ClassBasedComponent</div>;
  }
}
