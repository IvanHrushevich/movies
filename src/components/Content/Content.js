import React from 'react';
import PropTypes from 'prop-types';

import classes from './Content.module.scss';
import { MovieCard } from '../MovieCard/MovieCard';

export const Content = ({ movies }) => {
  const movieCards = movies.map((movie) => (
    <MovieCard
      src={movie.poster_path}
      title={movie.title}
      genre={movie.genre}
      year={movie.year}
      key={movie.id}
    />
  ));

  return (
    <main className={classes.section}>
      <div className={classes.cardsWrapper}>{movieCards}</div>
    </main>
  );
};

Content.propTypes = {
  movies: PropTypes.array,
};
