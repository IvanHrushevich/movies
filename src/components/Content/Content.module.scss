@import '../../shared/variables.scss';

.container {
  flex-grow: 1;
  min-height: 60rem;
  padding: $container-padding;
  background-color: $content-section-color;
}
