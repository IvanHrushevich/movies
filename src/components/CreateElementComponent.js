import React from "react";

export class CreateElementComponent extends React.Component {
  render() {
    return React.createElement(
      "div",
      null,
      "Hello from CreateElementComponent"
    );
  }
}
