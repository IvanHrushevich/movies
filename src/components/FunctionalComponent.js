import React from 'react';

import classes from './FunctionalComponent.module';

export const FunctionalComponent = () => <div className={classes.block}>Hello from FunctionalComponent</div>;
