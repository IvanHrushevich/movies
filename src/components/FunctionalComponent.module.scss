$text-color: blue;

.block {
  color: $text-color;
}
