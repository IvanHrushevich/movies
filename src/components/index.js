export * from './Footer/Footer';
export * from './MovieCard/MovieCard';

export * from './UI/index';
