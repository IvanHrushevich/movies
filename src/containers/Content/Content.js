import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import classes from './Content.module.scss';
import { MovieCard } from '../../components/index';

const Content = ({ movies }) => {
  const movieCards = movies.map((movie) => (
    <MovieCard
      src={movie.poster_path}
      title={movie.title}
      genre={movie.genres.join(' ')}
      year={parseInt(movie.release_date)}
      key={movie.id}
    />
  ));

  return (
    <main className={classes.section}>
      <div className={classes.cardsWrapper}>{movieCards}</div>
    </main>
  );
};

Content.propTypes = {
  movies: PropTypes.array,
};

const mapStateToProps = (state) => ({ movies: state.movies });

export default connect(mapStateToProps)(Content);
