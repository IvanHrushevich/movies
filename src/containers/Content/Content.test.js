import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';

import Content from './Content';
import { MovieCard } from '../../components/index';
import { mockStore } from '../../test-utils/index';

describe('Content', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore;

    component = mount(
      <Provider store={store}>
        <Content />
      </Provider>
    );
  });

  it('should render properly', () => {
    expect(component).toMatchSnapshot();
  });

  // it('should contain proper number of MovieCard components', () => {
  //   const fakeMovies = [
  //     {
  //       src: 'src1',
  //       title: 'title1',
  //       genre: 'genre1',
  //       year: 'year1',
  //       id: 'id1',
  //     },
  //     {
  //       src: 'src2',
  //       title: 'title2',
  //       genre: 'genre2',
  //       year: 'year2',
  //       id: 'id2',
  //     },
  //   ];

  //   component.setProps({ movies: fakeMovies });
  //   expect(component.find(MovieCard)).toHaveLength(2);
  // });
});
