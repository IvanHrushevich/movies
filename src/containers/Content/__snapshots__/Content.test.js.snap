// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render properly 1`] = `
<main
  className="section"
>
  <div
    className="cardsWrapper"
  />
</main>
`;
